pkg/model/components/etcd.go

```diff
@@ -32,13 +34,31 @@ var _ loader.OptionsBuilder = &EtcdOptionsBuilder{}
 func (b *EtcdOptionsBuilder) BuildOptions(o interface{}) error {
 	spec := o.(*kops.ClusterSpec)
 
+	imageVersion := ""
 	// @check the version are set and if not preset the defaults
 	for _, x := range spec.EtcdClusters {
 		// @TODO if nothing is set, set the defaults. At a late date once we have a way of detecting a 'new' cluster
 		// we can default all clusters to v3
 		if x.Version == "" {
 			x.Version = "2.2.1"
 		}
+
+		if imageVersion != "" {
+			if imageVersion != x.Version {
```

Comment:
Here is the odd thing @gambol99 - I think we could technically end up with multiple different versions of etcd here.  Please tell me I am wrong.  I am wondering if we should move the version up a level in the structs.

Reply:
logic seems right ... on first iteration set and every other has to equal .. unless i'm missing something obvious  :-)

Reply:
initial the version was set in the outer struct, i recall @justinsb want it to be per etcd cluster though ..

Reply:
the multiple version was protected on the spec validation https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/validation/legacy.go#L473-L495

Reply:
If I am reading that code it does not validate it.

Reply:
Can we move this to validation?

Reply:
@gambol99 I will back out the code, can you guys move it to validation?  But, I am really confused why we have the version where it is.  Oh well :)  I can do that validation later, but I have a long long list of things on my plate.